plugins/copyfilelocation.lua

```diff
@@ -1,15 +1,35 @@
 -- mod-version:3
 local core = require "core"
 local command = require "core.command"
+local treeview = require 'plugins.treeview'
+local menu = treeview.contextmenu
 
-command.add("core.docview", {
-  ["copy-file-location:copy-file-location"] = function(dv)
-    local doc = dv.doc
+command.add("core.docview!", {
+  ["doc:copy-file-location"] = function(view)
+    local doc = view.doc
     if not doc.abs_filename then
       core.error "Cannot copy location of unsaved doc"
       return
     end
-    core.log("Copying to clipboard \"%s\"", doc.abs_filename)
+    core.log("Copying \"%s\" to clipboard", doc.abs_filename)
     system.set_clipboard(doc.abs_filename)
   end
 })
+
+command.add(function()
+  return treeview.hovered_item ~= nil
+end, {
+  ["treeview:copy-file-location"] = function()
+    local item = core.active_view.hovered_item
+    if not (item and item.abs_filename) then
```

Comment:
```suggestion
command.add(function()
  return treeview.hovered_item ~= nil, treeview.hovered_item
end, {
  ["treeview:copy-file-location"] = function(item)
    if not (item and item.abs_filename) then
```

Reply:
damn, that just works??

Reply:
Yep, neat system introduced to avoid having to possibly repeat the same work done by the predicate.

We really need docs...